physionet-django/annotation/tests.py

```diff
@@ -1,3 +1,366 @@
+        self.assertEqual(response["location"]["coord_system"], "pixels")
+        self.assertEqual(response["location"]["x"], 50)
+        self.assertEqual(response["location"]["y"], 100)
+        self.assertEqual(response["location"]["width"], 200)
+        self.assertEqual(response["location"]["height"], 150)
+
+    def test_create_annotation_timeseries_interval(self):
+        """
+        Test create annotation with timeseries interval location
+        """
+        self.collection = AnnotationCollection.objects.create(
+            slug="test-collection-timeseries-interval",
+            name="Test Collection Timeseries Interval",
+            description="Test Description",
+            created_by=self.user,
+        )
+        self.annotation_type = AnnotationType.objects.create(
+            slug="test-annotation-type-timeseries-interval",
+            name="Test Annotation Type Timeseries Interval",
+            description="Test Description",
+            label_schema={
+                "type": "object",
+                "properties": {
+                    "label": {"type": "string"},
+                    "confidence": {"type": "number", "minimum": 0.0, "maximum": 1.0},
+                },
+                "required": ["label"],
+            },
+            allowed_location_type="timeseries_interval",
+        )
+        timeseries_interval_annotation_data = {
+            "annotation_type": self.annotation_type.slug,
+            "project": self.project.slug,
+            "file_path": "../test-ecg-record.wfdb",
+            "labels": {"label": "Normal Sinus Rhythm", "confidence": 0.95},
+            "location": {
+                "location_type": "timeseries_interval",
+                "coord_system": "samples",
+                "channel": "II",
+                "start": 1000,
+                "end": 5000,
+            },
+        }
+        self.access_token = AccessToken.objects.create(
+            user=self.user,
+            scope="annotations:edit",
+            expires=timezone.now() + timedelta(seconds=300),
+            token="secret-access-token-key",
+            application=self.application,
+        )
+        self.auth_header = self._create_authorization_header(self.access_token.token)
+        response = self._create_annotation(data=timeseries_interval_annotation_data)
+        self.assertEqual(response.status_code, 201)
+        response = response.json()
+        self.assertEqual(response["file_path"], "../test-ecg-record.wfdb")
+        self.assertEqual(
+            response["labels"], {"label": "Normal Sinus Rhythm", "confidence": 0.95}
+        )
+        # Testing timeseries interval setting
+        self.assertEqual(response["location"]["location_type"], "timeseries_interval")
+        self.assertEqual(response["location"]["coord_system"], "samples")
+        self.assertEqual(response["location"]["channel"], "II")
+        self.assertEqual(response["location"]["start"], 1000)
```

Comment:
Add test to check for location type mismatch (i.e. we should raise an error if someone tries to create an annotation with a location type that doesn't match the `AnnotationType`'s `allowed_location_type`